app/views/posts/new.html.erb

```diff
@@ -0,0 +1,10 @@
+<div class="show_and_edit">
+  <%= form_for post do |f| %>
+    <%= show_errors_on_base(post, :base) %>
+    <%= render partial: "input", locals: { form: f, field_name: :title } %>
+    <%= render partial: "input", locals: { form: f, field_name: :url } %>
+    <p class="center">OR</p>
+    <%= render partial: "input", locals: { form: f, field_name: :text } %>
```

Comment:
could the repetition of `render partial: "input", locals: { form: f, field_name: FIELD_NAME }` be cleaned up?

Reply:
I'm confused by the direction, that you seemed to be hinting at earlier, when you mentioned `<%= input_tag f, field_name %>`. I thought the purpose of the `locals` hash is to pass an instance of the form builder and the attribute to the partial (I was looking at [http://guides.rubyonrails.org/layouts_and_rendering.html#using-partials](url) ). I'm not seeing how a helper could reduce whats in the locals hash - but maybe there is more flexibility allowed than what I have experience with.

Reply:
that's a fair point -- all the helper method is really buying you is saved keystrokes and the ability to move that partial in the future without the need to update a whole bunch of references in the code. There's also the extensibility of if you wanted to have a `placeholder` or `hint` text, then you simply make them arguments on the helper method with a sensible default. Using the `locals` makes that trickier to set a default -- you'd end up setting a local var in the template, which isn't the best place to set a Ruby variable.